parteh/PRTAllometricCarbonMod.F90

```diff
@@ -0,0 +1,981 @@
+  use FatesAllometryMod   , only : h_allom
+  use FatesAllometryMod   , only : CheckIntegratedAllometries
+  use FatesAllometryMod   , only : StructureResetOfDH
+
+  use FatesGlobals        , only : endrun => fates_endrun
+  use FatesGlobals        , only : fates_log
+  use shr_log_mod         , only : errMsg => shr_log_errMsg
+  use FatesConstantsMod   , only : r8 => fates_r8
+  use FatesConstantsMod   , only : i4 => fates_int
+  use FatesIntegratorsMod , only : RKF45
+  use FatesIntegratorsMod , only : Euler
+  use EDPftvarcon         , only : EDPftvarcon_inst
+  use FatesConstantsMod   , only : calloc_abs_error
+  use FatesConstantsMod   , only : nearzero
+  use FatesConstantsMod   , only : itrue
+
+
+  implicit none
+  private
+
+  ! -------------------------------------------------------------------------------------
+  !
+  ! Define the state variables for this specific hypothesis.
+  !
+  ! -------------------------------------------------------------------------------------
+
+  integer, parameter :: leaf_c_id   = 1   ! Unique object index for leaf carbon
+  integer, parameter :: fnrt_c_id   = 2   ! Unique object index for fine-root carbon
+  integer, parameter :: sapw_c_id   = 3   ! Unique object index for sapwood carbon 
+  integer, parameter :: store_c_id  = 4   ! Unique object index for storage carbon
+  integer, parameter :: repro_c_id  = 5   ! Unique object index for reproductive carbon
+  integer, parameter :: struct_c_id = 6   ! Unique object index for structural carbon
+  integer, parameter :: num_vars = 6      ! THIS MUST MATCH THE LARGEST INDEX ABOVE
+  
+  
+  ! For this hypothesis, we integrate dbh along with the other 6. Since this
+  ! is a boundary condition, we do not add it to the state array, but we do want
+  ! to include it with the integrator array.
+
+  integer, parameter :: dbh_id             = 7  ! This is just used for the integrator
+  integer, parameter :: n_integration_vars = 7
+
+
+  ! -------------------------------------------------------------------------------------
+  ! Boundary Conditions
+  ! -------------------------------------------------------------------------------------
+  ! Input Boundary Indices (These are public, and therefore
+  !       each boundary condition across all modules must
+  !       have a unique name !!!!)
+  ! -------------------------------------------------------------------------------------
+
+  integer, public, parameter :: ac_bc_inout_id_dbh   = 1   ! Plant DBH
+  integer, public, parameter :: ac_bc_inout_id_netdc = 2   ! Index for the net daily C input BC
+  integer, parameter         :: num_bc_inout         = 2
+
+
+  integer, public, parameter :: ac_bc_in_id_pft   = 1   ! Index for the PFT input BC
+  integer, public, parameter :: ac_bc_in_id_ctrim = 2   ! Index for the canopy trim function
+  integer, parameter         :: num_bc_in         = 2
+
+  ! THere are no purely output boundary conditions
+  integer, parameter         :: num_bc_out      = 0
```

Comment:
Can there be an output 'boundary condition'? There are outputs though, which makes this a little confusing. Does it need to be here?

Reply:
I would like to add some protections to the boundary conditions, which will ensure that input boundary coditions aren't allowed to be changed.  Also, it may enforce that output boundary conditions are un-initialized prior to this being called.  Thus, we differentiate the boundary conditions into those 3 classes.   We do however need to specify if there are no boundary conditions in any of the three.   
The CNP hypothesis (coming in a later PR) will utialize all 3 types.